bin/p2-exec/main.go

```diff
@@ -114,6 +114,24 @@ func main() {
 	if err != nil {
 		log.Fatal(err)
 	}
+
+	/*
+		from `man setpgid`:
+		* "If pid is zero, then the call applies to the current process" / "if pid is 0, the process ID of the calling process shall be used"
+		* "Also, if pgid is 0, the process ID of the indicated process shall be used."
+		So we set the current process's process group ID to be equal to that of its process ID.
+		Without this, its process group ID is equal to its parent's.
+		This means that operations that act on their entire process tree will affect not only their own pod,
+		but all pods on the same machine (in a typical setup, where they all originate from the same parent).
+
+		For example, a renice may change the entire parent process tree's nice level.
+		Worse, this new nice level may propagate into a container, causing the container to be unable to set its nice level back to 0.
+		This has been observed to cause containers to fail to start.
+	*/
+	if err := unix.Setpgid(0, 0); err != nil {
```

Comment:
I understand that this will start every process exec'd by p2-exec in its own pgroup, thus giving us certain desirable properties. 

I do not know of any verification that's been performed in the wild to verify that this sort of thing is safe. Suppose, for instance, an app is using p2-exec for its environment directory abilities and expecting (probably implicitly) that the result of this command will share a pgroup with their main app. Said another way, p2-exec is used in several places not only in runit /run files.